lldb/source/Plugins/Process/elf-core/RegisterContextPOSIXCore_arm64.cpp

```diff
@@ -339,6 +337,18 @@ bool RegisterContextCorePOSIX_arm64::ReadRegister(const RegisterInfo *reg_info,
       value.SetFromMemoryData(*reg_info, src + sizeof(sve::user_za_header),
                               reg_info->byte_size, lldb::eByteOrderLittle,
                               error);
+    } else if (m_register_info_up->IsSMERegZT(reg)) {
+      value.SetFromMemoryData(*reg_info, m_zt_data.GetDataStart(),
+                              reg_info->byte_size, lldb::eByteOrderLittle,
+                              error);
+    } else {
+      offset = reg_info->byte_offset - m_register_info_up->GetSMEOffset();
+      assert(offset < sizeof(m_sme_pseudo_regs));
```

Comment:
Since you're using an assert, I'll ask:
Is this a hard error that isn't recoverable from? Or could we do something else here?

Reply:
This would be someone trying to fetch a register that is at a greater offset than the register context buffer has.  My worry about the assert is more that in a non-debug/release build, there's no error checking, we'll read from past the end of the m_sme_psueo_regs buffer and copy it into value.
On the other hand, this does fall in to "this shouldn't ever happen", but maybe returning false here instead of filling in value would be the best choice.

Reply:
The asserts are mostly for people like us because they generally happen if you are adding new registers and didn't update something.

But you are right Jason about non-asserts builds.

Since this is existing code I'm going to be cheeky and land this patch, and come back with another PR that makes these
assert in debug builds and return false for release builds.

Reply:
https://github.com/llvm/llvm-project/pull/71175